src/org/zaproxy/zap/extension/invoke/InvokeAppWorker.java

```diff
@@ -101,7 +105,32 @@ protected Void doInBackground() throws Exception {
 						.replace("%port%", port)
 						.replace("%site%", site)
 						.replace("%cookie%", cookie)
-						.replace("%postdata%", postdata);
+						.replace("%postdata%", postdata)
+						.replace("%msgid%", String.valueOf(msgid));
+
+				// Replace header tags
+				Matcher headers = Pattern.compile("%header-([a-zA-z0-9_-]+)%").matcher(finalParameter);
```

Comment:
`a-z` could be removed (per `A-z`) (alternatively it could be used the flag to be case insensitive).